src/main/java/it/pagopa/pn/delivery/rest/PnNotificationInputController.java

```diff
@@ -35,29 +41,46 @@ public PnNotificationInputController(PnDeliveryConfigs cfgs, NotificationReceive
 
     @Override
     public ResponseEntity<NewNotificationResponse> sendNewNotification(String xPagopaPnUid, CxTypeAuthFleet xPagopaPnCxType, String xPagopaPnCxId, NewNotificationRequest newNotificationRequest, List<String> xPagopaPnCxGroups) {
-        NewNotificationResponse svcRes = svc.receiveNotification(xPagopaPnCxId, newNotificationRequest);
-        return ResponseEntity.ok( svcRes );
+        PnAuditLogBuilder auditLogBuilder = new PnAuditLogBuilder();
+        PnAuditLogEvent logEvent = auditLogBuilder.before(PnAuditLogEventType.AUD_NT_PRELOAD, "presignedUploadRequest")
+                .uid(xPagopaPnUid)
+                .cxId(xPagopaPnCxId)
+                .cxType(xPagopaPnCxType.toString())
+                .build();
+        NewNotificationResponse svcRes = new NewNotificationResponse();
+        try {
+            svcRes = svc.receiveNotification(xPagopaPnCxId, newNotificationRequest);
+            logEvent.generateSuccess().log();
+        } catch (PnInternalException exc) {
+            logEvent.generateFailure(exc.getMessage()).log();
+        }
+        return ResponseEntity.ok(svcRes);
     }
 
 
     @Override
-    public ResponseEntity<List<PreLoadResponse>> presignedUploadRequest(String xPagopaPnUid, CxTypeAuthFleet xPagopaPnCxType, String xPagopaPnCxId, List<PreLoadRequest> preLoadRequest) {
+    public ResponseEntity<List<PreLoadResponse>> presignedUploadRequest(
+            String xPagopaPnUid, CxTypeAuthFleet xPagopaPnCxType, String xPagopaPnCxId, List<PreLoadRequest> preLoadRequest) {
         Integer numberOfPresignedRequest = cfgs.getNumberOfPresignedRequest();
-        if ( preLoadRequest.size() > numberOfPresignedRequest ) {
-            log.error( "Presigned upload request lenght={} is more than maximum allowed={}",
-                    preLoadRequest.size(), numberOfPresignedRequest );
-            throw new PnValidationException("request",
-                    Collections.singleton( new ConstraintViolationImpl<>(
-                            String.format( "request.length = %d is more than maximum allowed = %d",
-                                    preLoadRequest.size(),
-                                    numberOfPresignedRequest))));
+        PnAuditLogBuilder auditLogBuilder = new PnAuditLogBuilder();
+        PnAuditLogEvent logEvent = auditLogBuilder.before(PnAuditLogEventType.AUD_NT_PRELOAD, "presignedUploadRequest")
+                .uid(xPagopaPnUid)
+                .cxId(xPagopaPnCxId)
+                .cxType(xPagopaPnCxType.toString())
+                .build();
+        if (preLoadRequest.size() > numberOfPresignedRequest) {
+            String logMessage = String.format("Presigned upload request lenght=%s is more than maximum allowed=%s",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;size&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 72. Consider formatting this data with &#39;%d&#39; instead.